specification/automation/resource-manager/Microsoft.Automation/preview/2022-06-30-preview/job.json

```diff
@@ -0,0 +1,999 @@
+          ],
+          "x-ms-enum": {
+            "name": "JobStreamType",
+            "modelAsString": true
+          }
+        },
+        "streamText": {
+          "type": "string",
+          "description": "Gets or sets the stream text."
+        },
+        "summary": {
+          "type": "string",
+          "description": "Gets or sets the summary."
+        },
+        "value": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "object"
+          },
+          "description": "Gets or sets the values of the job stream."
+        }
+      },
+      "x-ms-client-flatten": true,
+      "description": "Definition of the job stream."
+    },
+    "JobStreamListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobStream"
+          },
+          "description": "A list of job streams."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets or sets the next link."
+        }
+      },
+      "description": "The response model for the list job stream operation."
+    },
+    "Job": {
+      "type": "object",
+      "description": "Definition of the job.",
+      "x-ms-mutability": [
+        "read",
+        "create"
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/JobProperties",
+          "x-ms-client-flatten": true,
+          "description": "The properties of the job."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../common/v1/definitions.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "JobListResultV2": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobCollectionItem",
+            "readOnly": true
+          },
+          "description": "List of jobs."
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The  link to the next page."
+        }
+      },
+      "description": "The response model for the list job operation."
+    },
+    "JobCollectionItem": {
+      "description": "Job collection item properties.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Job properties.",
+          "$ref": "#/definitions/JobCollectionItemProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../common/v1/definitions.json#/definitions/ProxyResource"
+        }
+      ],
+      "required": [
+        "properties"
+      ]
+    },
+    "JobProperties": {
+      "type": "object",
+      "properties": {
+        "runbook": {
+          "$ref": "#/definitions/RunbookAssociationProperty",
+          "description": "Gets or sets the runbook."
+        },
+        "runtime": {
```

Comment:
>runtime [](http://example.com/codeflow?start=9&length=7)

Is this an Id? Is this a name of an existing resource? In case is an existing resource, are you adding proper validation to check if the resource exists? are you returning the appropriate error message?

Reply:
Yes it is the new runtime resource, renamed this property as runtimeId and the RP will validate if the resource id is accurate.